hub/util/cache_chain.py

```diff
@@ -0,0 +1,15 @@
+from hub.core.storage.lru_cache import LRUCache  # type: ignore
+from typing import List
+from hub.core.storage.provider import StorageProvider
+
+# TODO proper exceptions
+def get_cache_chain(cache_list: List[StorageProvider], size_list: List[int]):
+    if not cache_list:
+        raise Exception
```

Comment:
In case there's no caching, the cache_list will be of length 1 corresponding to the actual storage. An empty cache_list is problematic as it means there's no storage.